apps/VendingMachine/contracts/VendingMachine.ral

```diff
@@ -83,42 +79,37 @@ Contract VendingMachine(
     pub fn nftByIndex(index: U256) -> INFT {
         let nftTokenId = subContractId!(toByteVec!(index))
         assert!(contractExists!(nftTokenId), ErrorCodes.NotMintedYet)
-
         return INFT(nftTokenId)
     }
     
     // *  *  * Write Functions *  *  * // 
 
     @using(preapprovedAssets = true, updateFields = true, assetsInContract = true, checkExternalCaller = false)
-    pub fn mint(foodType: U256) -> ByteVec {
+    pub fn mint(foodTypeId_: U256,mintAmount: U256) -> ByteVec {
         assert!(!mintIsPaused, ErrorCodes.MintIsPaused)
-        assert!(totalSupply + 1 <= MaxSupply, ErrorCodes.MaxSupplyReached)
-        let minter = callerAddress!()
-
-        let mut contractId = #
-        let random_ = IRandomNumberGenerator(addressToContractId!(@xeCgiV9x2sFysJiubJZjviUoaCEkVofeBJeUh4B1EF5R)).getRandomNumber(toByteVec!(minter))
-        // if (foodType == 1){
-
-        // }else if (foodType == 2){
+        assert!(totalSupply + mintAmount <= MaxSupply, ErrorCodes.MaxSupplyReached)
+        assert!(mintAmount <= MaxMintPerTx, ErrorCodes.MaxPerTx)
 
-        // } else {
-            
-        // }
-
-        contractId = contractId ++ mint_{minter -> ALPH: 1 alph}(minter,  random_ + 1)
-
-
-        transferTokenToSelf!(minter, ALPH, MintPrice)
-
-        totalSupply = totalSupply + 1
-
-        emit NftMinted(minter, contractId, 1)
-        return toByteVec!(foodType) // return contract id later
+        let minter = callerAddress!()
+        let mut contractIds = #
+        let mut startingIndex = 0
+        let mut nftIdToMint = 0
+
+        for (let mut i = 0; i < mintAmount; i = i + 1) {
+            nftIdToMint = pickIdForFood(foodTypeId_)
```

Comment:
does this work if `mintAmount` is 10, but there are only 5 supply available? what happens?

Reply:
` pub fn mint(foodTypeId_: U256,mintAmount: U256) -> ByteVec {
        assert!(!mintIsPaused, ErrorCodes.MintIsPaused)
        assert!(totalSupply + mintAmount <= MaxSupply, ErrorCodes.MaxSupplyReached)
        assert!(mintAmount <= MaxMintPerTx, ErrorCodes.MaxPerTx)`
` assert!(totalSupply + mintAmount <= MaxSupply, ErrorCodes.MaxSupplyReached)`